### DISCUSSION TITLE
String schema

### DISCUSSION THREAD

**Discussion by GreenArrow18** [e3dio/packBytes#15] [Q&A] [answered]
From the API:
schema = string('str1', 'str2', ..) // any of specific strings, auto-maps to integer

How do I use this properly? I am getting errors when I specify a list of strings as above.

const bits = PackBytes.numberToBits(values.length - 1);
165 | 		const z = {
166 | 			bits,
167 | 			bytes: Math.ceil(bits / 8),
168 | 			index: values,
169 | 			values: values.reduce((obj, v, i) => ((obj[v] = i), obj), {}),
                        ^
TypeError: values.reduce is not a function. (In 'values.reduce((obj, v, i) => (obj[v] = i, obj), {})', 'values.reduce' is undefined)

**Comment by e3dio** [comment, marked as answer]
Docs needed to be updated, I fixed that and added more of the new types available:
```js
schema = string([ 'str1', 'str2', .. ]) // any of specific strings, auto-maps to integer

schema = varint // variable length integer (max integer = 1_073_741_823) 
schema = objectid // MongoDB ObjectID
schema = uuid // MongoDB BSON UUID object
schema = date // javascript Date
schema = lonlat // coordinates [ longitude, latitude ]
```

**Reply by GreenArrow18** [reply]
Thanks. Works great!

varint is also very useful, my schemas use even less bytes now.
